api/saml/configuration/model.py

```diff
@@ -64,6 +65,64 @@ class SAMLConfiguration(ConfigurationGrouping):
         format="narrow",
     )
 
+    patron_id_use_name_id = ConfigurationMetadata(
+        key="saml_patron_id_use_name_id",
+        label=_("Patron ID: Use SAML NameID"),
+        description=_(
+            "Boolean value indicating whether SAML NameID should be searched for a unique patron ID: "
+            "<br>"
+            "- <b>0</b> means that NameID won't be used, "
+            "<br>"
+            "- <b>1</b> (default) means that Circulation Manager will scan NameID in search for a patron ID."
+        ),
+        type=ConfigurationAttributeType.NUMBER,
+        required=False,
+        default=1,
+    )
+
+    patron_id_attributes = ConfigurationMetadata(
+        key="saml_patron_id_attributes",
+        label=_("Patron ID: SAML Attributes"),
+        description=_(
+            "List of SAML attributes that MAY contain a unique patron ID. "
+            "The attributes will be scanned sequentially in the order you chose them, "
+            "and the first existing attribute will be used to extract a unique patron ID."
+            "<br>"
+            "NOTE: If a SAML attribute contains several values, only the first be used."
+        ),
+        type=ConfigurationAttributeType.MENU,
+        required=False,
+        options=[
+            ConfigurationOption(attribute.name, attribute.name)
+            for attribute in SAMLAttributeType
+        ],
+        default=[
+            SAMLAttributeType.eduPersonUniqueId.name,
+            SAMLAttributeType.eduPersonTargetedID.name,
+            SAMLAttributeType.eduPersonPrincipalName.name,
+            SAMLAttributeType.uid.name,
+        ],
+        format="narrow",
+    )
+
+    patron_id_regular_expression = ConfigurationMetadata(
+        key="saml_patron_id_regular_expression",
+        label=_("Patron ID: Regular expression"),
+        description=_(
+            "Regular expression used to extract a unique patron ID from the attributes "
+            "specified in <b>Patron ID: SAML Attributes</b>. "
```

Comment:
The regular expression would also be applied to NameID, if found (in addition to the Patron ID: SAML Attributes), right?

Reply:
Yes, that's true, I updated the description.